bindings/go/oci/internal/resolver/resolver.go

```diff
@@ -0,0 +1,92 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+
+	"ocm.software/open-component-model/bindings/go/oci/spec"
+)
+
+const DefaultComponentDescriptorPathSuffix = "component-descriptors"
+
+func NewURLPathResolver(baseURL string) *CachingURLPathResolver {
+	return &CachingURLPathResolver{
+		BaseURL: baseURL,
+	}
+}
+
+// CachingURLPathResolver is a Resolver that resolves references to URLs for Component Versions and Resources.
+// It uses a BaseURL and a BaseClient to get a remote store for a reference.
+// each repository is only created once per reference.
+type CachingURLPathResolver struct {
+	BaseURL    string
+	BaseClient remote.Client
+	PlainHTTP  bool
+
+	DisableCache bool
+
+	cacheMu sync.RWMutex
+	cache   map[string]spec.Store
+}
+
+func (resolver *CachingURLPathResolver) SetClient(client remote.Client) {
+	resolver.BaseClient = client
+}
+
+func (resolver *CachingURLPathResolver) BasePath() string {
+	return resolver.BaseURL + "/" + DefaultComponentDescriptorPathSuffix
+}
+
+func (resolver *CachingURLPathResolver) ComponentVersionReference(component, version string) string {
+	return fmt.Sprintf("%s/%s:%s", resolver.BasePath(), component, version)
+}
+
+func (resolver *CachingURLPathResolver) Reference(reference string) (fmt.Stringer, error) {
+	return registry.ParseReference(reference)
```

Comment:
hmm, do we want to use `ParseReference()` or `LooseParseReference()` here?

Reply:
no, because this reference resolver actually needs to deal with URLs so there we have strict reference parsing